service/src/main/java/bio/terra/pipelines/app/controller/PipelineRunsApiController.java

```diff
@@ -160,6 +162,12 @@ public ResponseEntity<ApiAsyncPipelineRunResponse> getPipelineRunResult(
       throw new NotFoundException("Pipeline run %s not found".formatted(jobId));
     }
 
+    if (pipelineRun.getStatus().equals(CommonPipelineRunStatusEnum.PREPARING)) {
+      throw new BadRequestException(
```

Comment:
hmm, I disagree with this being a bad request / 400. it is an existing PipelineRun, it just has not been started yet. (calling the *job* endpoint should give a 400 since the *job* doesn't exist yet.) how about response code 201 ("[Created](https://restfulapi.net/http-status-codes/#2xx)") and keep the message you have here?

Reply:
discussed offline, the bad request response is ok here 👍